### PR TITLE
✨refactor: update to use new community ami account

### LINKED ISSUE #5161: Ensure e2e are passing before v2.7.0 release
/kind feature

**Describe the solution you'd like**

We need to ensure the e2e tests are passing before we do the v2.7.0 release.

We have merged a few changes without the e2e passing due to the AMI issue.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by richardcase**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Updates to the project to start using the new CNCF owned AWS account for publishing AMIs. The code has had the old VMWare account removed and the new account ID added. Additionally the docs have been updated including clarifying that the public AMIs should only be used for non-production usage.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relates #5131
Relates #4984 
Fixes #5161 

**Special notes for your reviewer**:

A new GitHub Actions workflow (see #5138 ) can be invoked to publish new images. Also #5169 will be done as a follow-up.

**Checklist**:

- [X] squashed commits
- [X] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
action required
A new community owned AWS account is now being used for publishing AMIs. The old account owner by a company is no longer accessible and the images are being automatically deleted by AWS. Support for ubuntu 18.04 & 20.04 has been dropped and ubuntu 24.04 is now the default OS if no OS is supplied.
```

### CODE REVIEW

File: docs/book/src/development/amis.md
Comment (nrb) [inline docs/book/src/development/amis.md:24]: How long should we support 20.04?

**Review by nrb — APPROVED** [review]
/lgtm

Looks good, one question on supported versions but it's not a blocker by any means.

**Comment by richardcase** [comment]
Until #5138 merges and AMIs are published:

/hold

**Comment by richardcase** [comment]
I'm hoping the e2e will pass now 🤞 

/test pull-cluster-api-provider-aws-e2e

**Comment by richardcase** [comment]
Excellent, its using the new AMIs:

```
  [1mSTEP:[0m Searching for AMI: name=capa-ami-ubuntu-24.04-v1.29.9* [38;5;243m@ 10/15/24 09:51:52.365[0m
  [1mSTEP:[0m Using AMI: image-id=ami-052cf6cadd89abb96 [38;5;243m@ 10/15/24 09:51:52.759[0m
```

**Comment by richardcase** [comment]
I have a suspicion as to why its failing some of the tests. Will try and work on it this afternoon (after a rebase )

File: docs/book/src/topics/images/built-amis.md
Comment (richardcase) [reply docs/book/src/topics/images/built-amis.md]: Changed the docs.

File: docs/book/src/topics/images/built-amis.md
Comment (richardcase) [reply docs/book/src/topics/images/built-amis.md]: I added a link to the issue.

**Comment by richardcase** [comment]
Getting the e2e tests passing after the AMI change is now blocked on #5165

**Comment by richardcase** [comment]
Seeing if it was a flake.

/test pull-cluster-api-provider-aws-e2e

**Comment by damdo** [comment]
It didn't kick in
/test pull-cluster-api-provider-aws-e2e

**Comment by damdo** [comment]
```
  [FAILED] Failed to create kind cluster "test-98a3ki": failed to pull image "kindest/node:v1.29.9": command "docker pull kindest/node:v1.29.9" failed with error: exit status 1
```

The most recent kindest/node available is v1.29.8 https://hub.docker.com/layers/kindest/node/v1.29.8/images/sha256-b69a150f9951ef41158ec76de381a920df2be3582fd16fc19cf4757eef0dded9?context=explore

**Comment by richardcase** [comment]
Thanks @damdo. I will update and try again.

**Comment by richardcase** [comment]
The e2e tests are passing 🎉 I have created 5169 to re-enable 3 tests that have been disabled.

**Review by damdo — COMMENTED** [review]
Great work here @richardcase 
Thanks for working hard on this!

/lgtm

/assign @AndiDog @dlipovetsky @Ankitasw @nrb 
For approval

**Comment by richardcase** [comment]
Also, should we cherry pick this so we can do a v2.6.2 release?

**Comment by damdo** [comment]
Yeah +1 for that

File: test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go
Comment (AndiDog) [inline test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:44]: Are those still not working or can they be changed back to `Describe` (not pending)? If they fail, can we add a comment?

[repeated]

Comment (damdo) [reply test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:44]: See https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/5169

File: docs/book/src/development/amis.md
Comment (richardcase) [reply docs/book/src/development/amis.md]: As discussed on slack we've dropped this, so just 22.04 and 24.04.

File: test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go
Comment (richardcase) [reply test/e2e/suites/unmanaged/unmanaged_CAPI_clusterclass_test.go:44]: I also created #5176 to review all other tests marked pending.

**Comment by damdo** [comment]
@richardcase does this one also close #5161  ?
If so, should we mark this PR as `fixes #5161` so it autocloses?

**Comment by richardcase** [comment]
> @richardcase does this one also close #5161 ? If so, should we mark this PR as `fixes #5161` so it autocloses?

Yes it does, thanks @damdo for thinking of that. I have updated the description.

**Pull request merged**
